util/template/template.go

```diff
@@ -0,0 +1,44 @@
+package template
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/valyala/fasttemplate"
+)
+
+const (
+	prefix = "{{"
+	suffix = "}}"
+)
+
+type Template interface {
+	Replace(replaceMap map[string]string, allowUnresolved bool) (string, error)
+}
+
+func New(s string) (Template, error) {
+	template, err := fasttemplate.NewTemplate(s, prefix, suffix)
+	if err != nil {
+		return nil, err
+	}
+	return &impl{template}, nil
+}
+
+type impl struct {
+	*fasttemplate.Template
+}
+
+func (t *impl) Replace(replaceMap map[string]string, allowUnresolved bool) (string, error) {
+	env := newEnv(envMap(replaceMap))
+	replacedTmpl := &bytes.Buffer{}
```

Comment:
Minor:
```suggestion
	var replacedTmpl *bytes.Buffer
```

Reply:
I need this to be non-pointer to keep my code simple when I return